ansible/ansible-ai-connect-service
# Use Django auth for upstream

Jira Issue: https://issues.redhat.com/browse/AAP-19354


## Description


- Setting env `DEPLOYMENT_MODE=upstream` will use Django auth to login user.  Plan to use:
`DEPLOYMENT_MODE=[saas|on-prem|upstream]` to distinguish `upstream` vs commercial `on-prem`.  Any other thoughts?
- The login page is plain but works
- (Not updating the terms/artwork in this PR)


## Testing

### Steps to test
1. Pull down the PR
2. set env `DEPLOYMENT_MODE=upstream`
3. run the deployment
4. expect to see `Creating a testing user for upstream mode...` like the following:

```
Operations to perform:
  Apply all migrations: admin, auth, contenttypes, db, oauth2_provider, organizations, sessions, social_django, users
Running migrations:
 ....
 ....
Creating a testing user for upstream mode...
Creating a new user testuser
Creating a new token called testuser_token
```
5. visit http://127.0.0.1 and click "Login" and use `testuser`/`testuser` to login

<img width="305" alt="image" src="https://github.com/ansible/ansible-wisdom-service/assets/2421248/26301f10-1fdb-4b54-ac58-70c62e71ebbc">

<img width="616" alt="image" src="https://github.com/ansible/ansible-wisdom-service/assets/2421248/10bdf26c-10c8-4b41-b127-440a3b87bc82">

### Scenarios tested


## Production deployment
